tests/integration/pilot/common/routing.go

```diff
@@ -207,13 +208,16 @@ spec:
         add:
           istio-custom-header: user-defined-value`,
 		opts: echo.CallOptions{
+			Retry:   echo.Retry{Options: []retry.Option{retry.Timeout(1 * time.Second)}},
+			Timeout: 10 * time.Millisecond,
 			Port: echo.Port{
 				Name: "http",
 			},
 			Count: 1,
 			Check: check.And(
 				check.OK(),
-				check.RequestHeader("Istio-Custom-Header", "user-defined-value")),
+				check.RequestHeader("Istio-Custom-Header", "user-defined-value"),
+				check.RequestHeader("intentionally-fail", "ensuring dump works")),
```

Comment:
cleanup

Reply:
I'm not sure if it works yet.. letting it run on CI since my machine is pretty busy rn